docs/source/quick.rst

```diff
@@ -0,0 +1,299 @@
+We do that with the following command::
+
+   $ python -m custodia.server quick.conf
+
+The server will output to the terminal logs about the operations being peformed
+against it.
+
+
+Testing
+=======
+
+Once the server is started we can move to another terminal to test it.
+To avoid some typing we'll create a shell alias::
+
+   $ alias curls='curl -s --unix-socket ./quick -H "REMOTE_USER: me"'
+
+Let's use curl to fetch the root object
+::
+
+   $ curls http://localhost/
+   {"message": "Quis custodiet ipsos custodes?"}
+
+The message "Quis custodiet ipsos custodes?" is emitted by the Root handler by
+default when querying the root ('/'). It is used to test that the basic
+pipeline is working and authorizing correctly.
+
+The Root handler automatically adds also a 'secrets' handler under the path
+'/secrets/', this is the actual basename of our secrets storage and will always
+be used going forward.
+
+Let's now create a new container for our secrets
+::
+
+   $ curls -X POST http://localhost/secrets/bucket/
+
+Be sure to always pass the trailing '/' chracter, or the server will assume you
+are trying to operate on a key rather than a container and will return you an
+error.
+
+Now that we have a container let's store a secret in the simplest way
+::
+
+   $ curls -H "Content-Type: application/octet-stream" -X PUT http://localhost/secrets/bucket/mykey -d 'P@ssw0rd'
+
+This command is telling the server that we want to store raw data (by passing
+the "Content-Type: application/octet-stream" header) in the secret named
+"mykey" in the container named "bucket". The content is the string "P@ssw0rd".
+NOTE: you must provide a Content-Type header or the operation will fail, the
+supported types are: application/json and application/octet-stream
+
+Let's now retrieve the secret we just stored
+::
+
+   $ curls -H "Accept: application/octet-stream" http://localhost/secrets/bucket/mykey
+   P@ssw0rd
+
+NOTE: when getting the header to use to indicate the Content-Type we want is
+"Accept: application/octet-stream", this follows the standard HTTP protocol.
+
+When the raw data method is used, the database will generally store data base64
+encoded, let's try to get the same data without specifying an accepted content
+type (which is the same as specifying "Accept: application/json")
+::
+
+   $ curls http://localhost/secrets/bucket/mykey
+   {"type":"simple","value":"UEBzc3cwcmQ="}
+
+NOTE: The value is the base64 encoding of the string "P@ssw0rd"
+
+Let's now try to list the contents of our container
+::
+
+   $ curls http://localhost/secrets/bucket/
+   ["mykey"]
+
+We are returned a json array with the list of available keys.
+
+Let's now remove this key
+::
+
+   $ curls -X DELETE http://localhost/secrets/bucket/mykey
+
+And list again our container
+::
+
+   $ curls http://localhost/secrets/bucket/
+   []
+
+Finally let's cleanup and remove the container too
+::
+
+   $ curls -X DELETE http://localhost/secrets/bucket/
+
+
+Adding Authentication
```

Comment:
Great addition, this session and the other below cover more advanced topics that makes Custodia very intersting to play with!